python_transport/wirepas_gateway/utils/argument_tools.py

```diff
@@ -363,6 +363,26 @@ def add_mqtt(self):
             ),
         )
 
+        self.mqtt.add_argument(
+            "--mqtt_retain_flag_supported",
+            default=os.environ.get("WM_SERVICES_MQTT_RETAIN_FLAG_SUPPORTED", True),
+            type=self.str2bool,
+            nargs="?",
+            const=True,
+            help=("Set to true if broker support retain flag"),
+        )
+
+        self.mqtt.add_argument(
+            "--mqtt_max_qos_supported",
+            default=os.environ.get("WM_SERVICES_MQTT_MAX_QOS_SUPPORTED", 2),
+            action="store",
+            type=self.str2int,
+            help=(
+                "Max qos supported by broker in case it has limitation"
```

Comment:
I would suggest something like: "Max QOS supported by the broker. This allows for automatically downgrading the QOS of a published message or a message received on a given topic."